build fails with web_support=0 for secure mqtt

I tried both master & dev, but the build fails with web_support=0 works fine with web support.
I'm trying to get secure mqtt to work which requires web to be disabled.
I'm using PLATFORMIO

Here is the log

Compiling .pioenvs\nodemcu-lolin\lib297\ESPAsyncTCP\AsyncPrinter.cpp.o
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino: In function 'bool _alexaBodyCallback(void*, uint8_t*, size_t, size_t, size_t)':
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:79:33: error: 'void*' is not a pointer-to-object type
alexa.createServer(false);
^
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:79:52: error: 'void*' is not a pointer-to-object type
alexa.createServer(false);
^
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:79:66: error: 'HTTP_GET' was not declared in this scope
alexa.createServer(false);
^
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:79:83: error: 'void*' is not a pointer-to-object type
alexa.createServer(false);
^
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino: In function 'bool _alexaRequestCallback(void*)':
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:87:27: error: 'void*' is not a pointer-to-object type
#if RELAY_PROVIDER == RELAY_PROVIDER_LIGHT
^
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:87:62: error: 'void*' is not a pointer-to-object type
#if RELAY_PROVIDER == RELAY_PROVIDER_LIGHT
^
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:89:33: error: 'void*' is not a pointer-to-object type
// Global switch
^
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:89:52: error: 'void*' is not a pointer-to-object type
// Global switch
^
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:89:66: error: 'HTTP_GET' was not declared in this scope
// Global switch
^
C:/Users/XXXXXX/espurna/code/espurna/alexa.ino:89:83: error: 'void*' is not a pointer-to-object type
// Global switch
^
Compiling .pioenvs\nodemcu-lolin\lib297\ESPAsyncTCP\ESPAsyncTCP.cpp.o

Tried disabling Alexa support with the following :
`
[env:nodemcu-lolin-ota-mqtt-ssl]
platform = ${common.platform}
framework = ${common.framework}
board = ${common.board_4m}
board_build.flash_mode = ${common.flash_mode}
lib_deps = ${common.lib_deps}
lib_ignore = ${common.lib_ignore}
build_flags = ${common.build_flags_4m1m} -DNODEMCU_LOLIN -DDEBUG_FAUXMO=Serial -DNOWSAUTH -DUSE_CUSTOM_H=1 -DWEB_SUPPORT=0 -DALEXA_SUPPORT=0 -DASYNC_TCP_SSL_ENABLED=1 -DMQTT_USE_ASYNC=1
upload_speed = ${common.upload_speed}
upload_port = ${common.upload_port}
upload_flags = ${common.upload_flags}
monitor_speed = ${common.monitor_speed}
extra_scripts = ${common.extra_scripts}
`
custom.h has admin password, wifi and mqtt settings

I get the following warning and build fails:

`Linking .pioenvs\nodemcu-lolin-ota-mqtt-ssl\firmware.elf
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o):(.bss.default_private_key+0x0): multiple definition of `default_private_key'
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o):(.bss.default_private_key+0x0): first defined here
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o):(.bss.default_private_key_len+0x0): multiple definition of `default_private_key_len'
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o):(.bss.default_private_key_len+0x0): first defined here
c:/users/v2ank/.platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld.exe: Warning: size of symbol `default_private_key_len' changed from 2 in .pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o) to 4 in .pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o)
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o):(.bss.default_certificate+0x0): multiple definition of `default_certificate'
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o):(.bss.default_certificate+0x0): first defined here
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o):(.bss.default_certificate_len+0x0): multiple definition of `default_certificate_len'
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o):(.bss.default_certificate_len+0x0): first defined here
c:/users/v2ank/.platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld.exe: Warning: size of symbol `default_certificate_len' changed from 2 in .pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o) to 4 in .pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o)
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o): In function `SSLContext::getIOContext(int)':
C:\Users\v2ank\.platformio\packages\framework-arduinoespressif8266@1.20300.1\libraries\ESP8266WiFi\src/WiFiClientSecure.cpp:183: multiple definition of `ax_port_read'
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o):c:\Users\v2ank\espurna\code/.piolibdeps\ESPAsyncTCP\src/tcp_axtls.c:545: first defined here
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o): In function `SSLContext::getIOContext(int)':
C:\Users\v2ank\.platformio\packages\framework-arduinoespressif8266@1.20300.1\libraries\ESP8266WiFi\src/WiFiClientSecure.cpp:183: multiple definition of `ax_port_write'
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o):c:\Users\v2ank\espurna\code/.piolibdeps\ESPAsyncTCP\src/tcp_axtls.c:489: first defined here
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o): In function `ax_get_file':
C:\Users\v2ank\.platformio\packages\framework-arduinoespressif8266@1.20300.1\libraries\ESP8266WiFi\src/WiFiClientSecure.cpp:568: multiple definition of `ax_get_file'
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o):c:\Users\v2ank\espurna\code/.piolibdeps\ESPAsyncTCP\src/tcp_axtls.c:470: first defined here
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib6e2\libESP8266WiFi.a(WiFiClientSecure.cpp.o): In function `ax_wdt_feed':
C:\Users\v2ank\.platformio\packages\framework-arduinoespressif8266@1.20300.1\libraries\ESP8266WiFi\src/WiFiClientSecure.cpp:612: multiple definition of `ax_wdt_feed'
.pioenvs\nodemcu-lolin-ota-mqtt-ssl\lib297\libESPAsyncTCP.a(tcp_axtls.c.o):c:\Users\v2ank\espurna\code/.piolibdeps\ESPAsyncTCP\src/tcp_axtls.c:584: first defined here
collect2.exe: error: ld returned 1 exit status
*** [.pioenvs\nodemcu-lolin-ota-mqtt-ssl\firmware.elf] Error 1
================================================================= [ERROR] Took 26.25 seconds =================================================================`


Please help

--- a/code/espurna/alexa.ino
+++ b/code/espurna/alexa.ino
@@ -36,14 +36,16 @@ void _alexaConfigure() {
     alexa.enable(wifiConnected() && alexaEnabled());
 }
 
-bool _alexaBodyCallback(AsyncWebServerRequest *request, uint8_t *data, size_t len, size_t index, size_t total) {
-    return alexa.process(request->client(), request->method() == HTTP_GET, request->url(), String((char *)data));
-}
+#if WEB_SUPPORT
+    bool _alexaBodyCallback(AsyncWebServerRequest *request, uint8_t *data, size_t len, size_t index, size_t total) {
+        return alexa.process(request->client(), request->method() == HTTP_GET, request->url(), String((char *)data));
+    }
 
-bool _alexaRequestCallback(AsyncWebServerRequest *request) {
-    String body = (request->hasParam("body", true)) ? request->getParam("body", true)->value() : String();
-    return alexa.process(request->client(), request->method() == HTTP_GET, request->url(), body);
-}
+    bool _alexaRequestCallback(AsyncWebServerRequest *request) {
+        String body = (request->hasParam("body", true)) ? request->getParam("body", true)->value() : String();
+        return alexa.process(request->client(), request->method() == HTTP_GET, request->url(), body);
+    }
+#endif
 
 #if BROKER_SUPPORT
 void _alexaBrokerCallback(const unsigned char type, const char * topic, unsigned char id, const char * payload) {
@@ -79,7 +81,7 @@ void alexaSetup() {
     moveSetting("fauxmoEnabled", "alexaEnabled");
 
     // Basic fauxmoESP configuration
-    alexa.createServer(false);
+    alexa.createServer(!WEB_SUPPORT);
     alexa.setPort(80);
 
     // Uses hostname as base name for all devices
@@ -116,6 +118,8 @@ void alexaSetup() {
 
     // Websockets
     #if WEB_SUPPORT
+        webBodyRegister(_alexaBodyCallback);
+        webRequestRegister(_alexaRequestCallback);
         wsOnSendRegister(_alexaWebSocketOnSend);
         wsOnReceiveRegister(_alexaWebSocketOnReceive);
     #endif
@@ -137,8 +141,6 @@ void alexaSetup() {
     });
 
     // Register main callbacks
-    webBodyRegister(_alexaBodyCallback);
-    webRequestRegister(_alexaRequestCallback);
     #if BROKER_SUPPORT
         brokerRegister(_alexaBrokerCallback);
     #endif

